apps/web/src/components/common/AddressBookInput/index.tsx

```diff
@@ -20,25 +20,26 @@ const abFilterOptions = createFilterOptions({
  *  Temporary component until revamped safe components are done
  */
 const AddressBookInput = ({ name, canAdd, ...props }: AddressInputProps & { canAdd?: boolean }): ReactElement => {
-  const addressBook = useAddressBook()
-  const { setValue, control } = useFormContext()
-  const addressValue = useWatch({ name, control })
   const [open, setOpen] = useState(false)
   const [openAddressBook, setOpenAddressBook] = useState<boolean>(false)
+  const mergedAddressBook = useAllMergedAddressBooks()
+
+  const { setValue, control } = useFormContext()
+  const addressValue = useWatch({ name, control })
 
-  const addressBookEntries = Object.entries(addressBook).map(([address, name]) => ({
-    label: address,
-    name,
+  const allAddressBookEntries = mergedAddressBook.map((entry) => ({
```

Comment:
can't we do this maping in the hook?

Reply:
We only need this mapping for rendering the list inside the `Autocomplete` while the hook is used in more places so I would keep it outside. We could probably put a `useMemo` around it though